ci-operator/step-registry/optional-operators/cvp-common/apply-secrets/optional-operators-cvp-common-apply-secrets-commands.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Steps for extracting and applying the kube_secrets ISV parameter
+# Expects the standard Prow environment variables to be set
+
+REHEARSAL_INSTALL_NAMESPACE="!create"
+# Example of valid pyxis URL: https://pyxis.api.redhat.com/v1/projects/certification/pid/ospid-55159875-7eb1-41b0-bd91-0638b73536f1
+OO_PYXIS_URL="${OO_PYXIS_URL:-""}"
+# The namespace into which the operator and catalog will be
+# installed. Special value `!create` means that a new namespace will be created.
+OO_INSTALL_NAMESPACE="${OO_INSTALL_NAMESPACE:-$REHEARSAL_INSTALL_NAMESPACE}"
+
+# Check if OO_PYXIS_URL exists, skip the whole step if not.
+if [[ -z "$OO_PYXIS_URL" ]]; then
+    echo "OO_PYXIS_URL is not defined, skipping step cvp-common-apply-secrets!"
+    exit 0
+else
+    echo "OO_PYXIS_URL is defined, proceeding with cvp-common-apply-secrets step."
+fi
+
+echo "Creating a new NAMESPACE"
+if [[ "$OO_INSTALL_NAMESPACE" == "!create" ]]; then
+    echo "OO_INSTALL_NAMESPACE is '!create': creating new namespace"
+    NS_NAMESTANZA="generateName: oo-"
+elif ! oc get namespace "$OO_INSTALL_NAMESPACE"; then
+    echo "OO_INSTALL_NAMESPACE is '$OO_INSTALL_NAMESPACE' which does not exist: creating"
+    NS_NAMESTANZA="name: $OO_INSTALL_NAMESPACE"
+else
+    echo "OO_INSTALL_NAMESPACE is '$OO_INSTALL_NAMESPACE'"
+fi
+
+if [[ -n "${NS_NAMESTANZA:-}" ]]; then
+    OO_INSTALL_NAMESPACE=$(
+        oc create -f - -o jsonpath='{.metadata.name}' <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  $NS_NAMESTANZA
+EOF
+    )
+fi
+
+# Creating file that contains namespace name
+echo "$OO_INSTALL_NAMESPACE" > "${SHARED_DIR}"/cvp-common-apply-secrets-namespace.txt
```

Comment:
suggestion: make the filename more generic, something like `operator-install-namespace.txt` so it makes sense if we/others create the namespace in different steps.

Reply:
Thanks for suggestion, I tend to assign too specific file names.